be/src/storage/rowset/bitmap_index_reader.cpp

```diff
@@ -21,22 +21,51 @@
 
 #include "storage/rowset/bitmap_index_reader.h"
 
+#include <bthread/sys_futex.h>
+
 #include <memory>
 
 #include "storage/range.h"
 #include "storage/types.h"
 
 namespace starrocks {
 
-Status BitmapIndexReader::load(FileSystem* fs, const std::string& file_name, const BitmapIndexPB* bitmap_index_meta,
-                               bool use_page_cache, bool kept_in_memory) {
-    _typeinfo = get_type_info(OLAP_FIELD_TYPE_VARCHAR);
-    const IndexedColumnMetaPB& dict_meta = bitmap_index_meta->dict_column();
-    const IndexedColumnMetaPB& bitmap_meta = bitmap_index_meta->bitmap_column();
-    _has_null = bitmap_index_meta->has_null();
+StatusOr<bool> BitmapIndexReader::load(FileSystem* fs, const std::string& filename, const BitmapIndexPB& meta,
+                                       bool use_page_cache, bool kept_in_memory) {
+    while (true) {
+        auto state = _state.load();
+        if (state == 2) {
+            return false;
```

Comment:
why return false? should be true/

Reply:
'false' means the index data was loaded by another caller.